ssl/statem/extensions.c

```diff
@@ -617,13 +619,18 @@ int tls_collect_extensions(SSL *s, PACKET *packet, unsigned int context,
          * itself handle unsolicited response checks.
          */
         if (idx < OSSL_NELEM(ext_defs)
-                && (context & (SSL_EXT_CLIENT_HELLO
+                && ((context & (SSL_EXT_CLIENT_HELLO
                                | SSL_EXT_TLS1_3_CERTIFICATE_REQUEST
                                | SSL_EXT_TLS1_3_NEW_SESSION_TICKET)) == 0
                 && type != TLSEXT_TYPE_cookie
                 && type != TLSEXT_TYPE_renegotiate
                 && type != TLSEXT_TYPE_signed_certificate_timestamp
-                && (s->ext.extflags[idx] & SSL_EXT_FLAG_SENT) == 0) {
+                && (s->ext.extflags[idx] & SSL_EXT_FLAG_SENT) == 0)
+#ifndef OPENSSL_NO_GOST
+                && !(context & SSL_EXT_TLS1_2_SERVER_HELLO
+                && type == TLSEXT_TYPE_cryptopro_bug)
```

Comment:
Remove both parentheses here

Reply:
Or oh wait...  these parentheses should probably be kept...

Reply:
I think that both pairs of parentheses should be kept because they surround complex conditions.

Reply:
Well, not quite.  Basically, you've changed `(A && B && C && D)` to `((A && B && C && D) && !(GOST && CONDITION))` where `(A && B && C && D && !(GOST && CONDITION))` is enough.

However, please indent the GOST condition properly.  It should be:

``` C
                && !(context & SSL_EXT_TLS1_2_SERVER_HELLO
                     && type == TLSEXT_TYPE_cryptopro_bug)
```

Reply:
Updated.